### ISSUE TITLE
[Training] Weight individual timestamps in forecasting window 

### ISSUE THREAD

**Issue by cchallu** [Nixtla/neuralforecast#632]
### Description

Current models learn to forecast the entire window of size `h`. This new feature will allow weighting individual timestamps of the forecasting window. A new parameter (for eg. `forecast_weights`), Tensor of size `h` would weigh the train and validation loss during training.

### Use case

* Give more importance to accurately forecasting particular regions (for example, traffic peak hours).
* Specialize models in learning an individual timestamp in the future, using `forecast_weights=[0,...,0, 1]`.

**Comment by marcopeix** [comment]
The parameter `horizon_weight` in different loss functions implements exactly this.
